Bottom: 35aac141c2713fbeb8df48e2953824e8b986df8e
Top:    7aa2c658e8fa8152dc761e7b585775befdc71b1f
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-09-06 00:00:34 +0100

Refresh of more-hack-to-avoid-gl-calls-wh

---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 3ea1e05..8ba7e76 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -475,12 +475,6 @@ ghid_set_color (hidGC gc, const char *name)
   double r, g, b, a;
   a = 1.0;
 
-  if (!check_gl_drawing_ok_hack)
-    {
-      current_color = NULL;
-      return;
-    }
-
   if (!alpha_changed && current_color != NULL)
     {
       if (strcmp (name, current_color) == 0)
@@ -501,6 +495,12 @@ ghid_set_color (hidGC gc, const char *name)
 
   gc->colorname = (char *) name;
 
+  if (!check_gl_drawing_ok_hack)
+    {
+      current_color = NULL;
+      return;
+    }
+
   if (gport->colormap == NULL)
     gport->colormap = gtk_widget_get_colormap (gport->top_window);
   if (strcmp (name, "erase") == 0)
